SU2_CFD/include/variables/CMeshVariable.hpp

```diff
@@ -0,0 +1,138 @@
+/*!
+ * \file CMeshVariable.hpp
+ * \brief Declaration and inlines of the class
+ *        to define the variables of the mesh movement.
+ * \author Ruben Sanchez
+ * \version 6.2.0 "Falcon"
+ *
+ * The current SU2 release has been coordinated by the
+ * SU2 International Developers Society <www.su2devsociety.org>
+ * with selected contributions from the open-source community.
+ *
+ * The main research teams contributing to the current release are:
+ *  - Prof. Juan J. Alonso's group at Stanford University.
+ *  - Prof. Piero Colonna's group at Delft University of Technology.
+ *  - Prof. Nicolas R. Gauger's group at Kaiserslautern University of Technology.
+ *  - Prof. Alberto Guardone's group at Polytechnic University of Milan.
+ *  - Prof. Rafael Palacios' group at Imperial College London.
+ *  - Prof. Vincent Terrapon's group at the University of Liege.
+ *  - Prof. Edwin van der Weide's group at the University of Twente.
+ *  - Lab. of New Concepts in Aeronautics at Tech. Institute of Aeronautics.
+ *
+ * Copyright 2012-2019, Francisco D. Palacios, Thomas D. Economon,
+ *                      Tim Albring, and the SU2 contributors.
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "CVariable.hpp"
+
+class CMeshVariable : public CVariable {
+protected:
+
+  unsigned short nDim;
+
+  su2double WallDistance;   /*!< \brief Store the wall distance in reference coordinates. */
+
+  su2double *Mesh_Coord;           /*!< \brief Store the reference coordinates of the mesh. */
```

Comment:
Isn't this equivalent to "Solution"? if it is I think you should use solution instead, saves you all the polymorphism and adding specific get/set methods.

Reply:
Nope, the Mesh_Coord variable stores the original coordinates of the mesh. The "Solution" vector is the displacement, so that the current coordinate of any mesh point is Mesh_Coord + Solution. At some point I had implemented it as "Displacement", but I changed it back for the exact same reason you mention here.